packages/next-on-pages/src/buildApplication/processVercelFunctions/dedupeEdgeFunctions.ts

```diff
@@ -193,21 +225,41 @@ async function buildFunctionFile(
 		return acc;
 	}, new Map<string, string>());
 
-	groupedImports.forEach((keys, path) => {
-		const relativeImportPath = getRelativePathToAncestor({
-			from: newFnLocation,
-			relativeTo: nopDistDir,
-		});
+	let chunkMapIdx = 0;
+	const chunksExportsMap = new Map<string, Set<string>>();
+
+	const relativeImportPath = getRelativePathToAncestor({
+		from: newFnLocation,
+		relativeTo: nopDistDir,
+	});
+
+	groupedImports.forEach((exports, path) => {
 		const importPath = normalizePath(
 			join(relativeImportPath, addLeadingSlash(path)),
 		);
 
-		functionImports += `import { ${keys} } from '${importPath}';\n`;
+		if (path.endsWith('.wasm')) {
+			// if we're dealing with a wasm file there is a single default export to deal with
+			const defaultExport = exports;
+			// we don't need/want to apply any code transformation for wasm imports
+			functionImports += `import ${defaultExport} from "${path}";`;
```

Comment:
suggest: what about making this a `string[]`

it's generally not recommended to += strings (not memory/cpu efficient)
you can `.join('\n')` after the loop to make that consistent.

Is it expected that the export map is not updated form wasm?

Reply:
Good point, updated thanks 🙂👍

> Is it expected that the export map is not updated for wasm?

do you mean the `chunksExportsMap`?
if so yes, that map is only for the chunks, where we apply the code transformation (that wraps the chunks in functions that we then call), wasm files are not part of this